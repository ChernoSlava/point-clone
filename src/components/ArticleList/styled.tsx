import styled from 'styled-components';

const ArticleListStyled = styled.div`
    padding: 0;
    margin: 0;
    max-width: 984px;
    padding: 24px;
    border-radius: 8px;
    background-color: rgb(255, 255, 255);
`;

const ArticleDateHeadingStyled = styled.h2`
    padding: 0;
    margin: 0;
`;

const ListStyled = styled.ul`
 padding: 0;
`;

export {
    ArticleListStyled,
    ArticleDateHeadingStyled,
    ListStyled,
}
