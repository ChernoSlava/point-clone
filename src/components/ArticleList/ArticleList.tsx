import React from 'react';
import { Article } from '../../types';
import { ArticleItem } from '../ArticleItem/ArticleItem';
import { ArticleDateHeading, ArticleListStyled, ArticleList as List } from './styled';

interface Props {
  articles: Article[];
}

const ArticleList: React.FC<Props> = ({ articles }) => {
  
  return (
    <ArticleListStyled>
      <ArticleDateHeading>Сегодня</ArticleDateHeading>
      <List>
        {articles.map((article: Article) => (
           <ArticleItem key={article.id} article={article} />      
        ))}
      </List>
    </ArticleListStyled>
  );
};

export { ArticleList };
