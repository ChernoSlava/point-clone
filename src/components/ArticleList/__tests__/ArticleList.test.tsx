import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import "@testing-library/jest-dom";

import { ArticleList } from "../ArticleList";

describe("ArticleList tests", () => {
  const articles = [
    {
      id: "1",
      title: "Article 1",
      url: "/article-1",
      image: "article-1.jpg",
      description: "Description for Article 1",
      thumbnail: "article-1.jpg",
      date: "2 дня назад",
    },
    {
      id: "2",
      title: "Article 2",
      url: "/article-2",
      image: "article-2.jpg",
      description: "Description for Article 2",
      thumbnail: "article-2.jpg",
      date: "8 дней назад",
    },
  ];

  test("renders article list correctly", () => {
    const { container, unmount } = render(
      <MemoryRouter>
        <ArticleList articles={articles} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    unmount();
  });
});
