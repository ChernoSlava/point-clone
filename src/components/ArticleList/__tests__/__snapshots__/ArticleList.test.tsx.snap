// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders article list correctly 1`] = `
<div>
  <div
    class="sc-fsYfdN fGPEJW"
  >
    <h2
      class="sc-qZrbh dvdZjT"
    >
      Сегодня
    </h2>
    <ul
      class="sc-jsEeTM jidTqx"
    >
      <a
        href="/article-1"
      >
        <li
          class="sc-beySPh jOqDaV"
        >
          <div
            class="sc-guDLey cdWRpq"
          >
            <img
              alt="Cartoon"
              class="sc-dmyCSP gfsvpd"
              src="article-1.jpg"
            />
          </div>
          <div
            class="sc-hLQSwg ezfliH"
          >
            <h3
              class="sc-eDLKkx bCcaMp"
            >
              Article 1
            </h3>
            <p
              class="sc-jTQCzO gawhwJ"
            >
              Description for Article 1
            </p>
            <p
              class="sc-iBdnpw giGobE"
            >
              <img
                alt="icon"
                class="sc-gLLuof gNHDaF"
                src="article-1.jpg"
              />
              2 дня назад
            </p>
          </div>
        </li>
      </a>
      <a
        href="/article-2"
      >
        <li
          class="sc-beySPh jOqDaV"
        >
          <div
            class="sc-guDLey cdWRpq"
          >
            <img
              alt="Cartoon"
              class="sc-dmyCSP gfsvpd"
              src="article-2.jpg"
            />
          </div>
          <div
            class="sc-hLQSwg ezfliH"
          >
            <h3
              class="sc-eDLKkx bCcaMp"
            >
              Article 2
            </h3>
            <p
              class="sc-jTQCzO gawhwJ"
            >
              Description for Article 2
            </p>
            <p
              class="sc-iBdnpw giGobE"
            >
              <img
                alt="icon"
                class="sc-gLLuof gNHDaF"
                src="article-2.jpg"
              />
              8 дней назад
            </p>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
